src/components/FlowRunGraphArtifactDrawer.vue

```diff
@@ -0,0 +1,113 @@
+<template>
+  <p-drawer
+    class="flow-run-graph-artifact-drawer"
+    :open="artifactId !== null"
+    placement="right"
+    @update:open="close"
+  >
+    <template v-if="artifactId && artifact">
+      <div class="flow-run-graph-artifact-drawer__header">
+        <div class="flow-run-graph-artifact-drawer__details">
+          <div v-if="artifact.key">
+            <span class="flow-run-graph-artifact-drawer__detail-title">Key</span>
+            <ArtifactKeyIconText :artifact-id="artifactId" />
+          </div>
```

Comment:
nit: This would make a nice component and that would make this component much easier to grok/read